### ISSUE TITLE
RTBF Auvio : list of videos is wrong (not filter by the chosen categories)

### ISSUE THREAD

**Issue by xaviermiller** [Catch-up-TV-and-More/plugin.video.catchuptvandmore#1414]
⚠️ *Please try the last beta version and check existing issues before issuing a bug.* ⚠️

### Describe the bug
A clear and concise description of what the bug is:

version 0.2.40-beta10
Kodi 21.1 on linux (arm64)

Since some weeks, impossible to get for example a movie. If I chose "Films", then the type of movie (e.g. "Comédie"), the resulting list is not what expected, this is as this the 200 first recents all items of RTBF Auvio, mostly the last recorded Radio and TV items.
The movies are sometimes mixed into this list

The player is working fine (I can play any presented item), the live section is OK

### To Reproduce
Steps to reproduce the behaviour:
1. go to "replay / Belgium / RTBF Auvio"
2. chose any category "eg: Films"
3. chose any subcategory "eg: Action"
4. the list of videos is not what expected and is always the same

### Expected behaviour
A clear and concise description of what you expected to happen: get the filtererd list of video items by category / subcategory

### Actual behaviour
Tell us what happens instead:

### Screenshots
If applicable, add screenshots to help explain your problem:

### Your environment
 - [ ] Android
 - [ ] iOS
 - [X] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Other

 - Operating system version/name: Gentoo Linux (intel 64 bits)
 - Kodi version: 21.1
 - Catch-up TV & More version: 0.2.40-beta10
 - Installation method: using the beta channel, automatic updates

### Additional context
Add any other context about the problem here:

### Debug log
If you can, please add your debug log information (Tutorial available here: https://kodi.wiki/view/Log_file/Easy):
![WhatsApp Image 2024-10-23 at 21 10 40](https://github.com/user-attachments/assets/330ec478-a48e-4fea-b409-2dd78f6ee575)

**Comment by xaviermiller** [comment]
log: 
`2024-10-23 21:17:45.265 T:3577     info <general>: Skipped 4 duplicate messages..
2024-10-23 21:17:45.265 T:3577  warning <general>: Setting most video properties through ListItem.setInfo() is deprecated and might be removed in future Kodi versions. Please use the respective setter in InfoTagVideo.
2024-10-23 21:17:51.882 T:3641     info <general>: Skipped 25 duplicate messages..
2024-10-23 21:17:51.882 T:3641    error <general>: DoWork - Direct texture file loading failed for resource://resource.images.catchuptvandmore/channels/be/brf_fanart.jpg
2024-10-23 21:17:52.515 T:3643    error <general>: DoWork - Direct texture file loading failed for resource://resource.images.catchuptvandmore/channels/be/vrt_fanart.jpg
2024-10-23 21:17:52.814 T:3641    error <general>: DoWork - Direct texture file loading failed for resource://resource.images.catchuptvandmore/channels/be/telemb_fanart.jpg
2024-10-23 21:17:53.231 T:3641    error <general>: DoWork - Direct texture file loading failed for resource://resource.images.catchuptvandmore/channels/be/rtc_fanart.jpg
2024-10-23 21:17:54.652 T:3577  warning <general>: Setting most video properties through ListItem.setInfo() is deprecated and might be removed in future Kodi versions. Please use the respective setter in InfoTagVideo.
2024-10-23 21:18:19.883 T:3577     info <general>: Skipped 42 duplicate messages..
2024-10-23 21:18:19.883 T:3577  warning <general>: ListItem.addStreamInfo() is deprecated and might be removed in future Kodi versions. Please use InfoTagVideo.addVideoStream(), InfoTagVideo.addAudioStream() and InfoTagVideo.addSubtitleStream().
2024-10-23 21:18:19.884 T:3577  warning <general>: Setting most video properties through ListItem.setInfo() is deprecated and might be removed in future Kodi versions. Please use the respective setter in InfoTagVideo.
2024-10-23 21:18:19.884 T:3577  warning <general>: ListItem.addStreamInfo() is deprecated and might be removed in future Kodi versions. Please use InfoTagVideo.addVideoStream(), InfoTagVideo.addAudioStream() and InfoTagVideo.addSubtitleStream().
2024-10-23 21:18:19.885 T:3577  warning <general>: Setting most video properties through ListItem.setInfo() is deprecated and might be removed in future Kodi versions. Please use the respective setter in InfoTagVideo.
2024-10-23 21:18:19.885 T:3577  warning <general>: ListItem.addStreamInfo() is deprecated and might be removed in future Kodi versions. Please use InfoTagVideo.addVideoStream(), InfoTagVideo.addAudioStream() and InfoTagVideo.addSubtitleStream().
2024-10-23 21:18:19.885 T:3577  warning <general>: Setting most video properties through ListItem.setInfo() is deprecated and might be removed in future Kodi versions. Please use the respective setter in InfoTagVideo.
2024-10-23 21:18:19.885 T:3577  warning <general>: ListItem.addStreamInfo() is deprecated and might be removed in future Kodi versions. Please use InfoTagVideo.addVideoStream(), InfoTagVideo.addAudioStream() and InfoTagVideo.addSubtitleStream().
2024-10-23 21:18:19.886 T:3577  warning <general>: Setting most video properties through ListItem.setInfo() is deprecated and might be removed in future Kodi versions. Please use the respective setter in InfoTagVideo.
2024-10-23 21:18:19.886 T:3577  warning <general>: ListItem.addStreamInfo() is deprecated and might be removed in future Kodi versions. Please use InfoTagVideo.addVideoStream(), InfoTagVideo.addAudioStream() and InfoTagVideo.addSubtitleStream().
[...]

`

**Comment by joaopa00** [comment]
I don't have a Belgian vpn. Only Belgian people may have a look at this.

**Comment by xaviermiller** [comment]
How can I help? I am software developer

**Comment by joaopa00** [comment]
The best way to help this project is to send patches.

For RTBF Auvio, you could have a look at the file resource/lib/channels/be/rtbf.py

**Comment by xaviermiller** [comment]
Sure will do. I've looked at this code file, I will check the parameters and URL's

**Comment by xaviermiller** [comment]
It seems the APIs are changed (_next), the search API is now 1.23. This will be complicated for me to understand how it woked before and how it should work. Will still try...

seems based on red bee media

**Comment by cvandeplas** [comment]
The list of videos is provided by the `list_videos_category()` function, which uses `URL_VIDEOS_BY_CAT_ID`.
That variable is using `v8` of the API. 

The [RTBF API documentation](https://www.rtbf.be/api/partner/generic?partner_key=) mentions `v9` is the official version and previous `v7`, `v8` are deprecated.
The documentation however does not give more details on how to use `v9`.  

Filtering using the `program_id` works fine, it's just the `category_id` that's broken in API v8.

**Comment by xaviermiller** [comment]
yes, seems categoryid is the problem...

list of categories : https://www.rtbf.be/api/partner/generic/embed/category?method=getTree&v=1&partnerID=
gives for example "Films / comédie" = >id=105

and the list by category, as documented in V8 gives wrong results:
https://www.rtbf.be/api/partner/generic/media/objectlist?v=9&category_id=105&target_site=mediaz&offset=0&limit=10&partner_key=

**Comment by xaviermiller** [comment]
meanwhile seems I've fixed the search function, and can then pick the video I want to watch...

**Comment by joaopa00** [comment]
Maybe you could share your work for interested people.

**Comment by xaviermiller** [comment]
> Maybe you could share your work for interested people.

Hi, I made a PR #1438

**Comment by cmoulliard** [comment]
Is the PR already released part of the latest catchup plugin and available on kodi ? Does it fix the issue reported ?

**Comment by xaviermiller** [comment]
> Is the PR already released part of the latest catchup plugin and available on kodi ? Does it fix the issue reported ?

No, this fixes only the seach feature. The other features of browsing are no more working due to API changes
